src/main/java/uk/gov/hmcts/reform/hmc/helper/PartyRelationshipDetailsMapper.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.hmc.helper;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.hmc.data.HearingPartyEntity;
+import uk.gov.hmcts.reform.hmc.data.PartyRelationshipDetailsEntity;
+import uk.gov.hmcts.reform.hmc.exceptions.BadRequestException;
+import uk.gov.hmcts.reform.hmc.model.PartyDetails;
+import uk.gov.hmcts.reform.hmc.model.RelatedParty;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class PartyRelationshipDetailsMapper {
+
+    public List<PartyRelationshipDetailsEntity> modelToEntity(PartyDetails partyDetails,
+                                                              List<HearingPartyEntity> hearingPartyEntities) {
+
+        List<RelatedParty> relatedParties = partyDetails.getIndividualDetails().getRelatedParties();
+        List<PartyRelationshipDetailsEntity> partyRelationshipDetails = new ArrayList<>();
+
+        if (relatedParties != null) {
+            for (RelatedParty relatedParty: relatedParties) {
+
+                String relatedPartyId = relatedParty.getRelatedPartyID();
+
+                final HearingPartyEntity targetTechParty =
+                        getHearingPartyEntityByReference(relatedPartyId, hearingPartyEntities);
+
+                final HearingPartyEntity sourceTechParty =
+                        getHearingPartyEntityByReference(partyDetails.getPartyID(), hearingPartyEntities);
```

Comment:
Could move this line above this `for` loop (still inside the `if`) so we only call it once.

Reply:
Fixed in https://github.com/hmcts/hmc-cft-hearing-service/pull/153/commits/768e22b379256ea506790b5fc818777349748c05